packages/component-submission/server/aggregates/Submission.js

```diff
@@ -2,21 +2,57 @@ const { intersection } = require('lodash')
 const { mergeObjects } = require('../utils')
 
 class Submission {
-  constructor({ models: { Manuscript, File, Team }, services: { Storage } }) {
+  constructor({
+    models: { Manuscript, File, Team, SemanticExtraction },
+    services: { Storage },
+  }) {
     this.ManuscriptModel = Manuscript
     this.FileModel = File
     this.TeamModel = Team
+    this.SemanticExtractionModel = SemanticExtraction
     this.Storage = Storage
+
+    if (this.ManuscriptModel == null) {
+      throw new Error('Cannot construct Submission without a Manuscript model')
+    }
+    if (this.FileModel == null) {
+      throw new Error('Cannot construct Submission without a File model')
+    }
+    if (this.TeamModel == null) {
+      throw new Error('Cannot construct Submission without a Team model')
+    }
+    if (this.SemanticExtractionModel == null) {
+      throw new Error(
+        'Cannot construct Submission without a SemanticExtraction model',
+      )
+    }
   }
 
   async initialize(manuscriptId, userId) {
+    // interface Extractions: {
+    //  id: string;
+    //  created: DateTime;
+    //  updated: DateTime;
+    //  fieldName: string;
+    //  value: string;
+    // }
+    const SemanticExtractionToModels = (acc, extraction) => ({
+      ...acc,
+      [extraction.fieldName]: extraction.value,
+    })
+
     this.manuscript = await this.ManuscriptModel.find(
       manuscriptId,
       userId,
       '[teams]',
     )
     this.files = await this.FileModel.findByManuscriptId(manuscriptId)
     this.teams = await this.TeamModel.findByManuscriptId(manuscriptId)
+    this.suggestions = (await this.SemanticExtractionModel.findByManuscriptId(
+      manuscriptId,
+    ))
+      .sort((exA, exB) => exA.updated > exB.updated)
```

Comment:
Ooo - I never thought about this... if we have multiple suggestions for the same field at different times - it makes sense that the latest is used. But should this logic exist on the client rather than the server?

Reply:
Should we be returning all suggestions to the client or just the most recent one? I'm not sure at the moment it would matter to be sending it all back but in future if we're extracting abstracts and other chunks of text it might not be best to send all semantic extractions back to the client, just those to be used.

Reply:
Is there a case where we would have multiple suggestions ?

I'm guessing there could be with a score, similar as when doing reverse geocoding in say google maps api and maybe let the client choose ?

Reply:
We've just discussed this here - but this is an interesting point. I'm inclined to make the server dumb for now and send all the suggestions back. But we need to perhaps think a little deeper about this and how it is to work in the future. I think there is a valid case of two different suggestions being valid for the same field - as well as two different suggestions representing an newer and older (i.e valid and invalid) ... Maybe the way to distinguish is to use a `method` so that only the recent suggestion from the same method is the most valid?

/cc @hdrury1 @petereast @de-code

Reply:
@diversemix I wasn't quite sure whether it affects how you use ScienceBeam? Maybe you could quickly explain it tomorrow.

Reply:
Can do ... @petereast I think the MVP of this is what we already discussed and is moving the logic into the client.